components/server/src/auth/auth-provider-service.ts

```diff
@@ -56,77 +63,279 @@ export class AuthProviderService {
+                ownerId: ap.ownerId,
+                organizationId: ap.organizationId,
+                verified: ap.verified,
+                host: ap.host,
+                icon: ap.icon,
+                hiddenOnDashboard: ap.hiddenOnDashboard,
+                disallowLogin: ap.disallowLogin,
+                description: ap.description,
+                scopes: this.toScopes(ap),
+                requirements: this.toScopeRequirements(ap),
+            };
+
+        const result: AuthProviderInfo[] = [];
+        for (const p of authProviders) {
+            const identity = user.identities.find((i) => i.authProviderId === p.id);
+            if (identity) {
+                result.push(toInfo(p));
+                continue;
+            }
+            if (p.ownerId === user.id) {
+                result.push(toInfo(p));
+                continue;
+            }
+            if (builtinAuthProvidersConfigured && !this.isBuiltIn(p)) {
+                continue;
+            }
+            if (this.isNotHidden(p) && this.isVerified(p)) {
+                result.push(toInfo(p));
+            }
+        }
         return result;
     }
+    /**
+     * This is extracted from auth provider handlers in order to avoid a
+     * dependency to host contexts.
+     *
+     * TODO(at) these defaults of provider types needs to be centralized.
+     */
+    private toScopeRequirements(entry: AuthProviderEntry) {
+        switch (entry.type) {
+            case "GitHub":
+                return {
+                    default: GitHubScope.Requirements.DEFAULT,
+                    publicRepo: GitHubScope.Requirements.PUBLIC_REPO,
+                    privateRepo: GitHubScope.Requirements.PRIVATE_REPO,
+                };
+            case "GitLab":
+                return {
+                    default: GitLabScope.Requirements.DEFAULT,
+                    publicRepo: GitLabScope.Requirements.DEFAULT,
+                    privateRepo: GitLabScope.Requirements.REPO,
+                };
+            case "Bitbucket":
+                return {
+                    default: BitbucketOAuthScopes.Requirements.DEFAULT,
+                    publicRepo: BitbucketOAuthScopes.Requirements.DEFAULT,
+                    privateRepo: BitbucketOAuthScopes.Requirements.DEFAULT,
+                };
+            case "Bitbucket":
+                return {
+                    default: BitbucketServerOAuthScopes.Requirements.DEFAULT,
+                    publicRepo: BitbucketServerOAuthScopes.Requirements.DEFAULT,
+                    privateRepo: BitbucketServerOAuthScopes.Requirements.DEFAULT,
+                };
+        }
+    }
+    private toScopes(entry: AuthProviderEntry) {
+        switch (entry.type) {
+            case "GitHub":
+                return GitHubScope.All;
+            case "GitLab":
+                return GitLabScope.All;
+            case "Bitbucket":
+                return BitbucketOAuthScopes.ALL;
+            case "Bitbucket":
```

Comment:
BBS

Reply:
nice catch, solved in 7ea303fd617e6bd6766af53117a5448a499ec812